### PR TITLE
Issue43218: Allow MultiAuthConfigs to specify "dictateFieldVisibility"

### PR DESCRIPTION
**Pull request by RosalineP**
#### Rationale
We now have a second checkbox on the LDAP authentication configuration page ("Read LDAP attributes") that needs to reveal additional properties when checked. The original code was only configured to reveal additional properties for one checkbox—this update allows for any number of 'dictateFieldVisibility' checkboxes.

#### Related Pull Requests
* https://github.com/LabKey/ldap/pull/46

#### Changes
* Determine whether a field should be displayed according to relevant field's 'dictateFieldVisibility' setting

### CODE REVIEW

File: core/src/client/components/DynamicConfigurationModal.tsx [hunk 1]
```diff
@@ -31,7 +31,8 @@ export default class DynamicConfigurationModal extends PureComponent<Props, Part
         const {authConfig, modalType} = this.props;
         const fieldValues: any = {};
         this.props.modalType.settingsFields.forEach(field => {
-            fieldValues[field.name] = field.name in this.props.authConfig ? this.props.authConfig[field.name] : field.defaultValue;
+            const value = this.props.authConfig[field.name] != null ? this.props.authConfig[field.name] : "";
```
Comment (RosalineP) [inline core/src/client/components/DynamicConfigurationModal.tsx]: Avoids React warning about rendering `null` in a text field

Comment (labkey-alan) [reply core/src/client/components/DynamicConfigurationModal.tsx]: Yeah I really wish react would just do this for you.

File: core/src/client/components/DynamicConfigurationModal.tsx [hunk 2]
```diff
@@ -66,7 +67,10 @@ export default class DynamicConfigurationModal extends PureComponent<Props, Part
         }
 
         Object.keys(this.state.fieldValues).map(item => {
-            form.append(item, this.state.fieldValues[item]);
+            const itemValue = this.state.fieldValues[item];
+            if (itemValue !== null && itemValue !== "") {
```
Comment (labkey-alan) [inline core/src/client/components/DynamicConfigurationModal.tsx]: You may want to do `itemValue.trim() !== ''`

**Review by labkey-alan — APPROVED** [review]
Looks good, left one minor note.

File: core/src/client/components/DynamicConfigurationModal.tsx (hunk 2 above)
Comment (RosalineP) [reply core/src/client/components/DynamicConfigurationModal.tsx]: Good point; done!

**Pull request merged**